flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1198,16 +1205,146 @@ class DidNotConverge(Exception):
     """
 
 
-class BlockDeviceCalculatorTests(SynchronousTestCase):
+class _WriteVerifyingExternalClient(object):
     """
-    Tests for ``BlockDeviceCalculator``.
+    Representation of an external client of the Flocker API. This client can be
+    used to try to write to paths on the system and verify that the writes are
+    persisted to the underlying blockdevice.
+
+    :ivar _known_mountpoints: Known mountpoints of the dataset. Once flocker
+        has returned a path to an external agent, there is no promising how
+        long that agent might cache that information before it tries to write
+        to that location. This agent keeps track of all mountpoints it has ever
+        seen and tries to write to all of them.
+    :ivar _device_path: The device path of the blockdevice that backs the
+        dataset. This is used to verify that writes are persisted to the
+        underlying blockdevice.
+    :ivar _dataset_id: The dataset_id of the dataset under test.
+    :ivar _node_uuid: The node_uuid of the node that is being tested.
+    :ivar _testing_mountroot: A mountroot to construct moutpoints under.
+    :ivar _blockdevice_manager: An ``IBlockDeviceManager`` provider that can be
+        used to mount block devices.
     """
-    def setUp(self):
-        self.deployer = create_blockdevicedeployer(self)
 
-    def teardown_example(self, token):
+    def __init__(self, dataset_id, node_uuid, mountroot, blockdevice_manager):
+        self._known_mountpoints = set()
+        self._device_path = None
+        self._dataset_id = dataset_id
+        self._node_uuid = node_uuid
+        self._testing_mountroot = mountroot
+        self._blockdevice_manager = blockdevice_manager
+
+    def _has_file_named(self, filename):
         """
-        Cleanup after running a hypothesis example.
+        Returns whether the blockdevice at self._device_path has a file with
+        the given name.
+        """
+        test_mountpoint = self._testing_mountroot.child(str(uuid4()))
+        test_mountpoint.makedirs()
+        try:
+            self._blockdevice_manager.mount(self._device_path, test_mountpoint)
+            result = test_mountpoint.child(filename).exists()
+            self._blockdevice_manager.unmount(self._device_path)
+            test_mountpoint.remove()
```

Comment:
You probably want the `remove()` in a `finally:`  clause, so it gets removed even if mount failed. Otherwise next time around `makedirs()` might fail.